[CI] GenerativeIT test failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #8916 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/brbuwl2uubtnw)
- [elasticsearch-periodic-platform-support #8886 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/7rfhxhnikx3sg)
- [elasticsearch-pull-request #76777 / part-3](https://gradle-enterprise.elastic.co/s/cqpuarsfmbykg)
- [elasticsearch-periodic-platform-support #8838 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/yvs5j3vi3yw5q)
- [elasticsearch-periodic-platform-support #8838 / ubuntu-2004_platform-support-unix](https://gradle-enterprise.elastic.co/s/2viltoql2fun2)
- [elasticsearch-pull-request #76287 / part-3](https://gradle-enterprise.elastic.co/s/cll6aug6yi7pw)
- [elasticsearch-intake #24214 / part3](https://gradle-enterprise.elastic.co/s/aamzylxff5amg)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT.test" -Dtests.seed=8CD5FDCA0172C39D -Dtests.locale=ku-Latn-TR -Dtests.timezone=America/Costa_Rica -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: from languages_lookup | eval  NfkCkoVIk = -4749296, language_code = -1784723826 | eval  `language_name` = true, language_code = "uB", YObRhwLf = null, sRiDnyxMa = 932098778, `language_name` = false, `language_name` = -2683260997669172325 | grok language_code "%{WORD:kkNQKIHlnWNu} %{WORD:jvgRmdraPkFN}" | dissect kkNQKIHlnWNu "%{LfhowpyojQv} %{mfETCPIVpOE}"| rename NfkCkoVIk as language_code | lookup join languages_lookup on language_code
error: Expecting at least [10] columns, got [8]
```

**Issue Reasons:**
- [main] 7 failures in test test (1.9% fail rate in 374 executions)
- [main] 2 failures in step part-3 (2.1% fail rate in 97 executions)
- [main] 3 failures in pipeline elasticsearch-periodic-platform-support (27.3% fail rate in 11 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (2.1% fail rate in 96 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -577,9 +577,6 @@ tests:
 - class: org.elasticsearch.qa.verify_version_constants.VerifyVersionConstantsIT
   method: testLuceneVersionConstant
   issue: https://github.com/elastic/elasticsearch/issues/125638
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/129819
 - class: org.elasticsearch.index.engine.ThreadPoolMergeExecutorServiceDiskSpaceTests
   method: testAbortingOrRunningMergeTaskHoldsUpBudget
   issue: https://github.com/elastic/elasticsearch/issues/129823

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/EsqlQueryGenerator.java
@@ -184,7 +184,7 @@ public class EsqlQueryGenerator {
             };
         }
         // all types
-        name = randomName(previousOutput);
+        name = randomBoolean() ? randomStringField(previousOutput) : randomNumericOrDateField(previousOutput);
         if (name == null) {
             return "count(*)";
         }

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/command/pipe/LookupJoinGenerator.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/command/pipe/LookupJoinGenerator.java
@@ -55,8 +55,15 @@ public class LookupJoinGenerator implements CommandGenerator {
         }
 
         // the -1 is for the additional RENAME, that could drop one column
-        if (previousColumns.size() - 1 > columns.size()) {
-            return new ValidationResult(false, "Expecting at least [" + previousColumns.size() + "] columns, got [" + columns.size() + "]");
+        int prevCols = previousColumns.size() - 1;
+
+        if (previousColumns.stream().anyMatch(x -> x.name().equals("<all-fields-projected>"))) {
+            // known bug https://github.com/elastic/elasticsearch/issues/121741
+            prevCols--;
+        }
+
+        if (prevCols > columns.size()) {
+            return new ValidationResult(false, "Expecting at least [" + prevCols + "] columns, got [" + columns.size() + "]");
         }
         return VALIDATION_OK;
     }

